lib/dialects/abstract/query-generator.js

```diff
@@ -41,8 +41,9 @@ class QueryGenerator {
   extractTableDetails(tableName, options) {
     options = options || {};
     tableName = tableName || {};
+    const schema = tableName.schema || options.schema || options.schema !== false && this.sequelize.options.searchPath || 'public';
```

Comment:
Is there a reason that `this.sequelize.options.searchPath` is used instead of `this.sequelize.options.schema`?

Also, thanks! We just spent an hour looking at this and came up with basically the same solution.

Reply:
it seems like `schema` and `searchPath` are somewhat similar but then used for different aspects of the Sequelize.... not sure exactly why. If I remember correctly, specifying `schema` only was not sufficient

Reply:
I agree with @PanopticaRising -- this function is utility in that it takes names and adds in the scope (schema) primarily for queries.  It takes a best guess at what that scope is and sets that as default.  In my testing, I've discovered it shouldn't default to `'public'`, which only applies to one specific database -- I have a PR that updates this line (https://github.com/sequelize/sequelize/pull/14578/files#r889673513)

The `searchPath` in postgres can be more than one schema. It is like a `$PATH` environment variable, that will check the locations for the bare resource you're requesting.  Which means if you have `SELECT * FROM users;` and 20 different tables called `users`, it's going to walk through each schema listed on `search_path` until it finds it.

This function is "dumb" in that it doesn't hit the database at all.  It doesn't know which schema the unqualified tableName would belong to.  If your search_path is `etl, test, qa, $USER, public`, how would Sequelize know which one to choose?

The answer: whatever is calling the function needs to be smarter if their table is that ambiguous.